752 Table of Contents

GCHQ-Developer-299 opened the pull request:
<!-- 🙏 Thank you for your contribution, it is greatly appreciated! -->

## Summary of the changes
Initial implementation of Table of Contents as a canary component, developed by @cd3859 and finished off by @GCHQ-Developer-299 

## Related issue
#752 

## Bugs i'm aware of

- [x] Theme prop doesn't seem to work (and dark mode cypress test fails accessibility)
- [x] You can't set maximumHeadingLevel below 2 or it breaks
- [x] Truncate=false causes toc-items to overlap horribly with their siblings
- [x] Truncation ellipsis ... only applies on large screensizes, doesn't work when TOC is positioned 'top' either. Tooltip doesn't appear at any point.
- [x] Position "top" doesn't change the TOC's position dynamically in storybook, requiring a full page refresh.
- [x] Logic that renders a tooltip when TOC-item is truncated doesn't seem to work.
- [x] The scrolling sidebar isn't tall enough when size is 'large' 
- [x] At zoom levels other than 100%, the TOC doesn't vertically offset correctly (this is a Storybook issue apparently. Sticky position doesn't work correctly for Page Header either for example. it's all good when you open in new tab) 
- [x] Clicking a TOC item doesn't move focus to the chosen heading
- [x] The TOC appears after the page body in the tab order, which is incorrect. 
- Modifying maximumHeadingLevel causes unusual behaviour with the sidebar pill (i don't know whether this is a priority to fix - is anyone gonna be updating that setting dynamically? And it's fixed as soon as the user clicks an item and the page updates)

## Checklist

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on.
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render. (MOSTLY - see "bugs i'm aware of")

evenstensberg commented on packages/canary-react/src/component-tests/IcTableOfContents/IcTableOfContents.cy.tsx:
```diff
@@ -0,0 +1,116 @@
+    });
+  });
+
+  it("should render nested", () => {
```
```suggestion
  it("should render nested elements", () => {
```

evenstensberg reviewed (commented):
just a small nitpick, nice work!

GCHQ-Developer-530 commented on packages/canary-web-components/src/components/ic-table-of-contents-item/ic-table-of-contents-item.tsx:
Could we update this component to match our usual template? Aware it's only an internal component, but there's no prop descriptions etc.

GCHQ-Developer-299 replied:
As discussed internally, i'll instead move toc-item inside the overall toc folder and keep it as a FunctionalComponent akin to ic-date-picker/ic-date-button

GCHQ-Developer-530 commented on packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.css:
```diff
@@ -0,0 +1,135 @@
+:host {
+  display: block;
+  border: none;
+  max-height: 100vh;
```
<img width="665" height="197" alt="Screenshot 2025-08-08 at 12 31 09" src="https://github.com/user-attachments/assets/27a63879-5440-4351-898c-faf777ba4fb4" />

Could we add a z-index to table of contents? I think that would fix the tooltip going behind the content.

GCHQ-Developer-530 commented on packages/canary-web-components/src/components/ic-table-of-contents-item/ic-table-of-contents-item.css:
```diff
@@ -0,0 +1,63 @@
+
+  border: none;
+  position: relative;
+  min-height: 1.5rem;
```
I think this min-height needs updating to match the Figma height. Looks like at the moment the total height (including padding) is 28, but it should be 32.
<img width="297" height="169" alt="Screenshot 2025-08-08 at 12 40 29" src="https://github.com/user-attachments/assets/003675f4-61d0-4e1f-8268-0a8ac328fb48" />
<img width="558" height="208" alt="Screenshot 2025-08-08 at 12 40 43" src="https://github.com/user-attachments/assets/443b751f-7616-4ba9-9b42-9dc2d9f8d603" />

GCHQ-Developer-299 replied:
Looking at the figma designs, it looks like the 32px height is for the Large variant.
Regular:
<img width="699" height="209" alt="Screenshot 2025-08-11 at 11 50 40" src="https://github.com/user-attachments/assets/fb96f8ee-abb8-47fa-a4aa-efbdcae2033e" />


Large:
<img width="694" height="165" alt="Screenshot 2025-08-11 at 11 50 54" src="https://github.com/user-attachments/assets/efab11b5-ae51-414b-b527-9c9e7c087246" />

GCHQ-Developer-530 reviewed (commented):
Looking good! Just noticed a couple of things

GCHQ-Developer-718 commented on packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx:
```diff
@@ -0,0 +1,361 @@
+  };
+
+  // generate list items from the headings in the page
+  private generateListItems(): HTMLElement[] {
```
It feels like nested ordered lists should be used when the heading level changes as a non-visual equivalent to the indent. As in:
```
<ol>
  <li><a>My h2</a></li>
    <ol>
      <li><a>My h3</a></li>
      <li><a>My other h3</a></li>
      <ol>
        <li><a>My h4</a></li>
      </ol>
    </ol>
  <li><a>My other h2</a></li>
  <li><a>Another h2</a></li>
</ol>
```

Maybe this is a question for a screen reader user though as I appreciate it could be annoying if there are a lot of nested lists with very few items, so maybe real users would prefer the list to be flattened. I'm just thinking that, if the heading level is important enough information to be communicated visually, it should have a non-visual alternative.

If users do prefer having a flattened list for quicker navigation, you could consider including the heading level in a visually hidden span so they still have access to that information. That's not something I've seen before though.

GCHQ-Developer-299 replied:
I am really struggling to work this out in the code, will reach out to the dev team to see if anyone can think of how to generate a nested list. I think it might need an extensive refactor

GCHQ-Developer-718 commented on packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx:
```diff
@@ -0,0 +1,361 @@
+    content?.style.setProperty("top", `${this.headerHeight}px`);
+  }
+
+  // add tooltip to the item if it is too wide
```
For some reason, the tooltip isn't appearing for me when a truncated list item is focused using the keyboard.

GCHQ-Developer-718 reviewed (commented):
Looks really good! I also just noticed a couple of things.

GCHQ-Developer-718 commented:
> > Could you add an extra story showing how the table of contents works with an inline skip link like tree view? Just to verify that everything looks and behaves okay.
> 
> @GCHQ-Developer-718 Because the table of contents wraps the page content, it's not suited to having a skip link put before or inside it right now (see screenshots). Adding skip link functionality would require adding a prop to conditionally render one in there and this is outside the Acceptance Criteria so I think it should be raised as a ticket for Feature Improvement
> 
> Skip link before treeview <img alt="Screenshot 2025-08-11 at 15 26 24" width="1303" height="283" src="https://private-user-images.githubusercontent.com/111292477/476610849-5f999a31-51ee-4061-9e2c-bd0acc6a4cee.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUwMDc5MDIsIm5iZiI6MTc1NTAwNzYwMiwicGF0aCI6Ii8xMTEyOTI0NzcvNDc2NjEwODQ5LTVmOTk5YTMxLTUxZWUtNDA2MS05ZTJjLWJkMGFjYzZhNGNlZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxMlQxNDA2NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNTMyNjM1Mzc2YjQwODQxZWI1ZTRjYWRjYTg2MWJjOTBjNDM3Y2JjY2ZkNmU2MDM4NDU4MGFjODc2M2NlY2Q5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.l60Id1qzZ0l8CKOxMUlLPYqjymMGC7UUBx2ogx1d0q0">
> 
> Skip link as child of treeview <img alt="Screenshot 2025-08-11 at 15 24 38" width="523" height="195" src="https://private-user-images.githubusercontent.com/111292477/476610909-07ee9a43-1107-4a03-b431-5c60146344be.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUwMDc5MDIsIm5iZiI6MTc1NTAwNzYwMiwicGF0aCI6Ii8xMTEyOTI0NzcvNDc2NjEwOTA5LTA3ZWU5YTQzLTExMDctNGEwMy1iNDMxLTVjNjAxNDYzNDRiZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxMlQxNDA2NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMGJmOTRkM2JlMTc2MWFiNjVlZWViMGYxZjM5Y2RiODdlYjZkNTg5OWUxMzllNWI1ZWFkMDg1N2RjNDRkMWQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MikVl6nDjgEGhhpFMcZdprDo_zekxHu9NbNDbFd5a8s">

Thanks for checking, I'm happy with a new ticket being raised. :)

GCHQ-Developer-718 reviewed (approved):
Looks great!

ad9242 commented:
This should not be merged until https://github.com/mi6/ic-design-system/issues/768 is complete

ad9242 commented:
seems to be an issue in the following story:

https://mi6.github.io/ic-ui-kit/branches/752-table-of-contents/canary-react/?path=/story/react-components-table-of-contents--default-top

ad9242 commented:
more comment than request for anything to be changed, but the tooltips seem a little redundant as they add no additional info, they just repeat the text:

<img width="354" height="125" alt="image" src="https://github.com/user-attachments/assets/5d32aa03-b599-4884-8d81-2620eaa1bcc0" />

GCHQ-Developer-299 commented:
> seems to be an issue in the following story:
> 
> https://mi6.github.io/ic-ui-kit/branches/752-table-of-contents/canary-react/?path=/story/react-components-table-of-contents--default-top

Fixed I think - this was because i'd used web-component syntax for the styles tags, switched it to JSX and it should be fine now

GCHQ-Developer-299 commented:
> Sorry another that is maybe more of a design decision: when the ToC is at the top, I feel like that part should be sticky, otherwise as soon as you scroll, you can no longer see it, so what is the point of it? I suppose it offers quick initial navigation, but after that you would have to return to the top to move between page sections
> 
> <img alt="image" width="950" height="500" src="https://private-user-images.githubusercontent.com/116102637/480433133-6b24ad2c-a631-47a9-a55f-615566cdc66f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3NzI0ODMsIm5iZiI6MTc1NTc3MjE4MywicGF0aCI6Ii8xMTYxMDI2MzcvNDgwNDMzMTMzLTZiMjRhZDJjLWE2MzEtNDdhOS1hNTVmLTYxNTU2NmNkYzY2Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyMVQxMDI5NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNzExNDIzM2I5NjcwNjI3MThhYWIzMGI3NGQ2MDA5YzU4M2UzMzM3ZDJhMDVlM2QwMDcyMjcyNTE4NTYwZDU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.NPQkUWofvES77fuAYXy60wNIMGe2uP3NGhWAQZDCxk8">

Looking at the ticket, the TOC scrolling with the page when positioned top is correct:

> Given that a table of contents exists on a page
When the table of contents is positioned to the top of the page
Then the table of contents scrolls with the page content.

Contrasted with side position:

> Given that a table of contents exists on a page
When the table of contents is positioned to the side of the page
Then the table of contents is sticky on the page content when scrolling

GCHQ-Developer-299 commented:
> > more comment than request for anything to be changed, but the tooltips seem a little redundant as they add no additional info, they just repeat the text:
> > <img alt="image" width="354" height="125" src="https://private-user-images.githubusercontent.com/116102637/480403116-5d32aa03-b599-4884-8d81-2620eaa1bcc0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3NjUyOTIsIm5iZiI6MTc1NTc2NDk5MiwicGF0aCI6Ii8xMTYxMDI2MzcvNDgwNDAzMTE2LTVkMzJhYTAzLWI1OTktNDg4NC04ZDgxLTI2MjBlYWExYmNjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyMVQwODI5NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZTNhZDI1M2JkN2U5MmQwNGZkYmRmNTk4NjM3ZjY1OTlhMWE1ZTM5YjFkZmMwYjQyMGY3YzE5NjEwNWUwZTM2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.M5PG8rMwIcC5J9kWo0evgQ_IkBfvIQMu3og3BmZTmck">
> 
> @ad9242 that is a bug though - the tooltips shouldn't be showing when the text isn't truncated.

@ad9242 i've addressed this with a fix to icTooltip's hide method. Should now behave as expected :)

ad9242 commented on packages/canary-web-components/src/utils/types.ts:
```diff
@@ -301,3 +301,5 @@ export enum IcWeekDays {
   Saturday = 6,
 }
+
+export type IcPosition = "top" | "right";
```
Can we make this more specific, eg IcPositionTopOrRight or even IcTableOfContentsPosition? IcPosition sounds like a name that could be useful more generally across other components in future.  This way it would not be a breaking change if we want to have an IcPosition further down the line which also supports "bottom" or "left"

ad9242 commented on packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx:
```diff
@@ -0,0 +1,442 @@
+  /**
+   * Whether the table of contents will appear on the right of or above child content on large viewport widths.
+   */
+  @Prop() position: IcPosition = "right";
```
i think this needs to be `mutable` as it complains about the value being changed in the unit tests

GCHQ-Developer-847 commented on packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx:
```diff
@@ -0,0 +1,442 @@
+    return (
+      <Host class={{ [`ic-theme-${theme}`]: theme !== "inherit" }}>
+        <ic-section-container
+          id={"table-of-contents-section-container"}
+          aligned={"full-width"}
```
I think these can just be:
```suggestion
          id="table-of-contents-section-container"
          aligned="full-width"
```

GCHQ-Developer-847 commented on packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx:
```diff
@@ -0,0 +1,442 @@
+            ${position === "right" && "table-of-contents-position-side"}`}
+          >
+            <ic-typography variant="subtitle-large">{heading}</ic-typography>
+            <div class={"table-of-contents-list"}>
```
```suggestion
            <div class="table-of-contents-list">
```

GCHQ-Developer-847 commented on packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.css:
```diff
@@ -0,0 +1,139 @@
+/*   --ic-table-of-contents-text: var(--ic-color-text-primary);
```
Can these commented-out lines be removed?

GCHQ-Developer-847 commented on packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx:
```diff
@@ -0,0 +1,442 @@
+                class={`table-of-contents-sidebar ${
+                  monochrome && `table-of-contents-sidebar-monochrome`
+                }`}
+                style={{ height: size === "large" ? "2rem" : "1.75rem" }}
```
Could this styling be moved into the CSS file instead? As we generally don't add inline styling like this in other components

Same with line 412

GCHQ-Developer-847 reviewed (changes requested):
Looks amazing 👏 Noticed a few things but they're only very small :)

GCHQ-Developer-530 commented:
I've come across an issue when trying to add table of contents guidance to our site, I thought it was just a component preview issue, but I think it'll be an issue when we come to replace the anchor nav with table of contents too so it might be worth trying to fix. 

Because they are h2s they're showing up in the table of contents, but because we change them to wrapped h2s, the table of contents labels include the div and permalink etc.

<img width="839" height="364" alt="Screenshot 2025-08-26 at 14 42 38" src="https://github.com/user-attachments/assets/f58db3f0-e5fe-4a37-b836-b74f891826ee" />

Is there a way to make sure it's only the h2 text content that makes the label?

GCHQ-Developer-847 commented on packages/canary-react/src/stories/ic-table-of-contents.stories.js:
```diff
@@ -0,0 +1,263 @@
+};
+
+/**
+ * The table of contents component is used to create a navigable list of sections within a page or document.
+ * 
+ * It can be positioned on the left, right or on top of the child content.
```
I think we're just putting all canary guidance onto the website now, rather than in Storybook, following [#1557](https://github.com/mi6/ic-design-system/issues/1557). So I guess this comment can be removed?

Also the comment in the web components Storybook. Although happy for the comments to be left  if you think they're still useful - but just wanted to point out that it will probably just be our team who will see them :)

GCHQ-Developer-847 reviewed (commented):
Looks really good :)

Just spotted one more thing, however. Also I think there are still a couple of the others issues left - the blue selected indicator doesn't stretch to the height of the table of contents item, and it's not visible in high contrast mode

<img width="266" height="123" alt="image" src="https://github.com/user-attachments/assets/cd773f9c-27bd-446b-9c24-dce0038a8c10" />

<br />

<img width="247" height="84" alt="image" src="https://github.com/user-attachments/assets/d5a5f664-42a1-4df6-90b8-8fd02641bb5d" />

Will be happy to approve after these are resolved

GCHQ-Developer-299 commented:
> Looks really good :)
> 
> Just spotted one more thing, however. Also I think there are still a couple of the other issues left - the blue selected indicator doesn't stretch to the height of the table of contents item, and it's not visible in high contrast mode
> 
> <img alt="image" width="266" height="123" src="https://private-user-images.githubusercontent.com/111882109/483556809-cd773f9c-27bd-446b-9c24-dce0038a8c10.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY0NzMzMzksIm5iZiI6MTc1NjQ3MzAzOSwicGF0aCI6Ii8xMTE4ODIxMDkvNDgzNTU2ODA5LWNkNzczZjljLTI3YmQtNDQ2Yi05YzI0LWRjZTAwMzhhOGMxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyOVQxMzEwMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YzZjY2Q2YTA4ODc0YjA4ZWUzZGYyMzY2YmRlMzFjZTE0MDc3ZTRkNzkzOTZiMWJkZmNlYTJlYTg1ZmJiMmVmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6H7hKODafJXEXPHGRTfru8r0Gd5B7mLQ-ENcTeWyl5k">
> <img alt="image" width="247" height="84" src="https://private-user-images.githubusercontent.com/111882109/483556963-d5a5f664-42a1-4df6-90b8-8fd02641bb5d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY0NzMzMzksIm5iZiI6MTc1NjQ3MzAzOSwicGF0aCI6Ii8xMTE4ODIxMDkvNDgzNTU2OTYzLWQ1YTVmNjY0LTQyYTEtNGRmNi05MGI4LThmZDAyNjQxYmI1ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyOVQxMzEwMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NjE0MDhiYjYwNjBmMzI0MjA4NWI5Yjk0OTM4MWZhYTkxZmQ4NzY3YTI3NWVjNWIzMGIxNzFjZGViYzcwNjJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3Yq-bbKa6Zj9xYlfJxsCWfuQUGhM5F9FN3btRPLbeVQ">
> Will be happy to approve after these are resolved

@GCHQ-Developer-847 thanks for the really thorough reviews this week - i've addressed the remaining issues with the sidebar - let me know if "highlight" isn't the correct colour for use in HC :)

GCHQ-Developer-847 commented:
> > Looks really good :)
> > Just spotted one more thing, however. Also I think there are still a couple of the other issues left - the blue selected indicator doesn't stretch to the height of the table of contents item, and it's not visible in high contrast mode
> > <img alt="image" width="266" height="123" src="https://private-user-images.githubusercontent.com/111882109/483556809-cd773f9c-27bd-446b-9c24-dce0038a8c10.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY0NzMzMzksIm5iZiI6MTc1NjQ3MzAzOSwicGF0aCI6Ii8xMTE4ODIxMDkvNDgzNTU2ODA5LWNkNzczZjljLTI3YmQtNDQ2Yi05YzI0LWRjZTAwMzhhOGMxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyOVQxMzEwMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YzZjY2Q2YTA4ODc0YjA4ZWUzZGYyMzY2YmRlMzFjZTE0MDc3ZTRkNzkzOTZiMWJkZmNlYTJlYTg1ZmJiMmVmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6H7hKODafJXEXPHGRTfru8r0Gd5B7mLQ-ENcTeWyl5k">
> > <img alt="image" width="247" height="84" src="https://private-user-images.githubusercontent.com/111882109/483556963-d5a5f664-42a1-4df6-90b8-8fd02641bb5d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY0NzMzMzksIm5iZiI6MTc1NjQ3MzAzOSwicGF0aCI6Ii8xMTE4ODIxMDkvNDgzNTU2OTYzLWQ1YTVmNjY0LTQyYTEtNGRmNi05MGI4LThmZDAyNjQxYmI1ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyOVQxMzEwMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NjE0MDhiYjYwNjBmMzI0MjA4NWI5Yjk0OTM4MWZhYTkxZmQ4NzY3YTI3NWVjNWIzMGIxNzFjZGViYzcwNjJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3Yq-bbKa6Zj9xYlfJxsCWfuQUGhM5F9FN3btRPLbeVQ">
> > Will be happy to approve after these are resolved
> 
> @GCHQ-Developer-847 thanks for the really thorough reviews this week - i've addressed the remaining issues with the sidebar - let me know if "highlight" isn't the correct colour for use in HC :)

No problem! :) I would agree that "highlight" is the most appropriate colour to use 👍 

Just noticed a small issue with the height of the selected indicator now though - when you change the `position` prop / change the screen size, the indicator doesn't adjust to match the new height (as the amount of lines the text wraps onto changes). It only updates after a page refresh

https://github.com/user-attachments/assets/773c0909-702f-4125-87d7-f87e4e00de1f
